docs/DeveloperGuide.md

```diff
@@ -137,55 +137,131 @@ Classes used by multiple components are in the `seedu.clinic.commons` package.
+    input into a valid `DeleteCommand` via `DeleteCommandParser`. <br>
+    
+    Checks for compulsory prefixes and valid arguments (`ct/TYPE` and `i/INDEX` in this case) are done. The code will throw a `ParseException` if the check fails.<br>
+    
+    If multiple entries of `ct/TYPE` or `i/INDEX` are found, the last entry will be used as the argument. <br>
+    
+    Afterwards, all the valid arguments (`INDEX` and `TYPE`) will form a `DeleteCommand`, which will be executed.
+    
+2. Execution <br>
+
+    The `DeleteCommand` is executed via a `execute` call from `LogicManager`. <br>
+    
+    Using the `targetType` attribute, the execution is first classified as either Supplier deletion (`s`) or Warehouse deletions (`w`). <br>
+    
+    Base on the classification, the model will retrieve the relevant displayed list of warehouse/supplier via `model#getFilteredWarehouseList()`/`model#getFilteredSupplierList()`. <br>
+    
+    It then locates the warehouse/supplier entry that user wants to delete via the `INDEX` passed in.
+    
+    Afterwards, `model#deleteWarehouse`/`model#deleteSupplier` will remove the target entry from the list in the `model`, and the `model` will then update the displayed list.
+    
+3. Result display <br>
+    With the deletion completed, a `CommandResult` will be returned to the `LogicManager` with a success message, which will
+    be shown to the user in the UI.
+
+![Delete Command Sequence 2 Diagram](images/DeleteCommandSequenceDiagram2.png)
+<center><i>Figure n. Delete Command Sequence Diagram for product deletion</i></center>
+<br>
+
+1. Parsing
+
+    The parsing workflow is the same except that now an additional field `pd/PRODUCT_NAME` will be checked (with both prefix and argument) and parsed. <br>
+    
+    Afterwards, all the valid arguments (`INDEX`, `TYPE` and `PRODUCT_NAME`) will form a `DeleteCommand`, which will be executed.
+    
+2. Execution
+
+    The `DeleteCommand` is executed via a `execute` call from `LogicManager`. <br>
+    
+    Using the `targetType` attribute, the execution is now classified as either Supplier-related product deletion (`ps`) or Warehouse-related product deletions (`pw`). <br>
+    
+    Base on the classification, the model will again retrieve the relevant displayed list of warehouse/supplier via `model#getFilteredWarehouseList()`/`model#getFilteredSupplierList()`. <br>
+    
+    It then locates the warehouse/supplier entry from whom the product to delete via the `INDEX` passed in.
+    
+    Next, the product matching the required `PRODUCT_NAME` will be retrieved via `getProductByName`, and an updated target entry with this product removed will be returned through `removeProduct` call.
+    
+    Afterwards, `model#setWarehouse`/`model#setSupplier` will replace the old entry with the updated target entry from the list in the `model`, and the `model` will then update the displayed list.
```

Comment:
Perhaps breaking this long sentence up would be clearer as well! "replace the old entry with the updated target entry from the list in `model` . The `model` will then update the displayed list." would be better :)

Reply:
Thanks!